### ISSUE TITLE
AtomConfig.ajax. jsonPostEncode not working

### ISSUE THREAD

**Issue by accessizer** [neurospeech/atoms.js#135]
I believe there is a problem with configuring jsonPostEncode.

In AtomPromise.js, we have the following:

    /*Line 282 - 'AtomPromise.js' */    var e = AtomConfig.ajax.jsonPostEncode;
    /*Line 283 - 'AtomPromise.js' */    if (e) {
    /*Line 284 - 'AtomPromise.js' */        o = e(o);
    /*Line 285 - 'AtomPromise.js' */    } else {
    /*Line 286 - 'AtomPromise.js' */        o.data = { formModel: JSON.stringify(o.data) };
    /*Line 287 - 'AtomPromise.js' */    }

When jsonPostEncode is set, the object `o` is totally overwritten with the function which returns the new data, as per the example provided in the documentation (section 17.2.3) when it should be writing to the `o.data`. As a result, the AtomPromise never returns (e.g. I see busy template indefinitely).

I modified and this seemed to work:

     /*Line 284 - 'AtomPromise.js' */        o.data = e(o.data, o);

However, this is not complete fix because the 2nd parameter of the function which is meant to set options and I believe would be merged into the object `o` which itself is the options does not get merged back into the object `o` (e.g. the `contentType` as suggested in the same documentation) do not get set at all.

**Comment by neurospeech** [comment]
Well, instead you can define a function such as, you can change multiple options
    
     function postEncode(o){
          o.data = JSON.stringify(o.data);
          o.type = 'application/json';
          o.lastModified = false;
     }

This is the reason, we had kept o = e(o), which allows you to change multiple changes to options. This option is then passed to $.ajax.

If you share your code, I can guide you on how to setup postEncode correctly.

**Comment by accessizer** [comment]
Thanks. I'm unfamiliar with github but for some reason it is not showing my NOTE edits that I added to the original post. After I posted the issue, I realized that I might have been following the documentation too blindly and as you said, modified my function to return the `o` object, rather than taking 2 arguments as originally documented in your mini-book. So I don't think it was actually a problem with the code but just the documentation.
